lib/services/reconciler.go

```diff
@@ -28,8 +28,15 @@ import (
 	"github.com/gravitational/teleport/api/types"
 )
 
+// Matcher is used by reconciler to match resources.
+type Matcher[T any] func(T) bool
+
 // ReconcilerConfig is the resource reconciler configuration.
 type ReconcilerConfig[T any] struct {
+	// NOTE: This struct *must* maintain the same layout as
+	//       GenericReconcilerConfig[string, T] in order for automatic
+	//       conversions to work
+
```

Comment:
```suggestion
// ReconcilerConfig is the resource reconciler configuration.
type ReconcilerConfig[T any] GenericReconcilerConfig[string, T]
```
this should work?

Reply:
Ah cute. I though I'd tried it before, but perhaps I tried an alias instead.